Streetcode/Streetcode.XUnitTest/MediatRTests/Media/Audio/CreateAudioTests.cs

```diff
@@ -0,0 +1,187 @@
+
+            // Act
+            var result = await handler.Handle(new CreateAudioCommand(testCreateAudioDTO), CancellationToken.None);
+
+            // Assert
+            Assert.True(result.IsFailed);
+            Assert.Equal(expectedError, result.Errors.First().Message);
+        }
+
+        [Fact]
+        public async Task ShouldThrowException_TitleIsRequired()
+        {
+            // Arrange
+            var expectedError = "Title is required";
+            _mockLocalizerFail.Setup(x => x["TitleIsRequired"])
+            .Returns(new LocalizedString("TitleIsRequired", expectedError));
+
+            var testCreateAudioDTO = GetCreateAudioDTO_WithoutTitle();
+            var testAudioDTO = GetAudioDTO();
+            var testAudio = GetAudio();
+
+            SetupCreateRepository(-1);
+            SetupBlobService();
+            SetupMapper(testAudio, testAudioDTO);
+
+            var handler = new CreateAudioHandler(_mockBlobService.Object, _mockRepository.Object, _mockLogger.Object, _mockMapper.Object, _mockLocalizerFail.Object);
+
+            // Act
+            var result = await handler.Handle(new CreateAudioCommand(testCreateAudioDTO), CancellationToken.None);
+
+            // Assert
+            Assert.True(result.IsFailed);
+            Assert.Equal(expectedError, result.Errors.First().Message);
+        }
+
+        private void SetupCreateRepository(int returnNumber)
+        {
+            _mockRepository.Setup(x => x.AudioRepository.Create(It.IsAny<AudioEntity>()));
+            _mockRepository.Setup(x => x.SaveChangesAsync()).ReturnsAsync(returnNumber);
+        }
+
+        private void SetupBlobService()
+        {
+            _mockBlobService.Setup(service => service.SaveFileInStorage(
+                It.IsAny<string>(),
+                It.IsAny<string>(),
+                It.IsAny<string>()))
+                .Returns("fake_blob_name");
+            _mockBlobService.Setup(service => service.FindFileInStorageAsBase64(
+                It.IsAny<string>()))
+                .Returns("fake_base64_string");
+        }
+
+        private void SetupMapper(AudioEntity testAudio, AudioDTO testAudioDTO)
+        {
+            _mockMapper.Setup(x => x.Map<AudioEntity>(It.IsAny<AudioFileBaseCreateDTO>()))
+                .Returns(testAudio);
+            _mockMapper.Setup(x => x.Map<AudioDTO>(It.IsAny<AudioEntity>()))
+                .Returns(testAudioDTO);
+        }
+
+        private static AudioFileBaseCreateDTO GetCreateAudioDTO()
+        {
+            return new AudioFileBaseCreateDTO()
+            {
+                Title = "Title",
+                MimeType = "string",
+                BaseFormat = "ab34",
+                Extension = "string",
+            };
+        }
+
+        private static AudioFileBaseCreateDTO GetCreateAudioDTO_WithoutExtension()
+        {
+            return new AudioFileBaseCreateDTO()
+            {
+                Title = "Title",
```

Comment:
this is okay, but the best practice is to randomly generate such variables or if the value in several variables is repeated, then put this value in a separate private variable